src/media-segment-request.js

```diff
@@ -164,6 +166,43 @@ const handleKeyResponse = (segment, objects, finishProcessingFn, triggerSegmentE
   return finishProcessingFn(null, segment);
 };
 
+/**
+ * Processes an mp4 init segment depending on the codec through the transmuxer.
+ *
+ * @param {Object} segment init segment to process
+ * @param {string} codec the codec of the text segments
+ */
+const initMp4Text = (segment, codec) => {
+  if (codec === WEB_VTT_CODEC) {
```

Comment:
nit: For both these functions, I would move the codec check on the call of the function, so it is not even called if it is not the correct codec.

Reply:
I think the idea for having these functions as generic `mp4Text` init/segments is to leave a low resistance path for other text codecs to be initialized/parsed, such as ttml. So we could do a `else if (codec === TTML_CODEC)` later. I'm not totally sold on that being necessary though, if you think it makes more sense to codec check outside we can remove these `mp4Text` functions altogether.

Reply:
I think it's fine as is if the expectation is that other checks will happen in this function